documentation/proposals/Proposal - Silk.NET 3.X Generic Maths.md

```diff
@@ -0,0 +1,884 @@
+    public static Vector2D<T> operator *(T left, Vector2D<T> right);
+
+    /// <summary>Divides the first vector by the second.</summary>
+    /// <param name="left">The first vector.</param>
+    /// <param name="right">The second vector.</param>
+    /// <returns>The vector that results from dividing <paramref name="left" /> by <paramref name="right" />.</returns>
+    /// <remarks>The <see cref="Vector2D{T}.op_Division" /> method defines the division operation for <see cref="Vector2D{T}" /> objects.</remarks>
+    public static Vector2D<T> operator /(Vector2D<T> left, Vector2D<T> right);
+
+    /// <summary>Divides the specified vector by a specified scalar value.</summary>
+    /// <param name="value1">The vector.</param>
+    /// <param name="value2">The scalar value.</param>
+    /// <returns>The result of the division.</returns>
+    /// <remarks>The <see cref="Vector2D{T}.op_Division" /> method defines the division operation for <see cref="Vector2D{T}" /> objects.</remarks>
+    public static Vector2D<T> operator /(Vector2D<T> value1, T value2);
+
+    /// <summary>Returns a value that indicates whether each pair of elements in two specified vectors is equal.</summary>
+    /// <param name="left">The first vector to compare.</param>
+    /// <param name="right">The second vector to compare.</param>
+    /// <returns><see langword="true" /> if <paramref name="left" /> and <paramref name="right" /> are equal; otherwise, <see langword="false" />.</returns>
+    /// <remarks>Two <see cref="Vector2D{T}" /> objects are equal if each element in <paramref name="left" /> is equal to the corresponding element in <paramref name="right" />.</remarks>
+    public static bool operator ==(Vector2D<T> left, Vector2D<T> right);
+
+    /// <summary>Returns a value that indicates whether two specified vectors are not equal.</summary>
+    /// <param name="left">The first vector to compare.</param>
+    /// <param name="right">The second vector to compare.</param>
+    /// <returns><see langword="true" /> if <paramref name="left" /> and <paramref name="right" /> are not equal; otherwise, <see langword="false" />.</returns>
+    public static bool operator !=(Vector2D<T> left, Vector2D<T> right);
+    #endregion
+
+    #region CopyTo
+    /// <summary>Copies the elements of the vector to a specified array.</summary>
+    /// <param name="array">The destination array.</param>
+    /// <remarks><paramref name="array" /> must have at least two elements. The method copies the vector's elements starting at index 0.</remarks>
+    /// <exception cref="NullReferenceException"><paramref name="array" /> is <see langword="null" />.</exception>
+    /// <exception cref="ArgumentException">The number of elements in the current instance is greater than in the array.</exception>
+    /// <exception cref="RankException"><paramref name="array" /> is multidimensional.</exception>
+    public void CopyTo(T[] array);
```

Comment:
Most of these instance methods are examples of where we changed them to be extension methods for `Vector64/128/256/512<T>` for the performance and other reasons given above.

For `ISimdVector` we likewise keep them static and implement it explicitly, having it defer to the extension method.

Reply:
Noted, will change.